docs/developer/crypto.md

```diff
@@ -1,9 +1,85 @@
 
-The controller looks for a cluster-wide private/public key pair on startup, and generates a new 4096 bit (by default) RSA key pair if not found. The key is persisted in a regular Secret in the same namespace as the controller. The public key portion of this (in the form of a self-signed certificate) should be made publicly available to anyone wanting to use SealedSecrets with this cluster. The certificate is printed to the controller log at startup, and available via an HTTP GET to /v1/cert.pem on the controller.
+## **Protocols and cryptograhic tools used**
 
-During encryption, each value in the original Secret is symmetrically encrypted using AES-GCM (AES-256) with a randomly-generated single-use 32 byte session key. The session key is then asymmetrically encrypted with the controller's public key using RSA-OAEP (using SHA256), and the original Secret's namespace/name as the OAEP input parameter (aka label). The final output is: 2 byte encrypted session key length || encrypted session key || encrypted Secret.
+Sealed-secrets uses the following protocols for the secrets management :
 
-Note that during decryption by the controller, the SealedSecret's namespace/name is used as the OAEP input parameter, ensuring that the SealedSecret and Secret are tied to the same namespace and name.
+- **AES-256-GCM** with a randomly generated single-use 32 bytes session key. Since the key is single-use, we do not use any nonce.
+- **RSA-OAEP**, with **SHA-256**. It is used to assure the confidentiality of the AES-256-GCM session key.
+- **X509** certificates are used to manage RSA public keys. This public key contained in the certificate can be used to encrypt AES-256-GCM session key.
 
-When using the namespace-wide scope, the OAEP input (aka label) only contains the namespace and in cluster-wide scope the label is an empty string.
+Certificates generated by the sealed secrets controller are renewed every 30 days and have a 10 years validity span.
+
+## **Functionning**
+
+### P**ublic/private key pair management**
+
+The controller looks for a cluster-wide private/public key pair on startup. If no key pair is found and none is provided manually, the controller generates a new 4096 bit (by default) RSA key pair. In both cases, the key pair is persisted in a regular Secret in the same namespace as the controller.
+
+The public key (in the form of a self-signed certificate if it was generated by the controller) should be made publicly available to anyone wanting to use SealedSecrets with this cluster.
+
+Note that it is possible to use your own X509 certificate with the command bellow :
+
+```
+kubeseal --cert [https:/]/path/to/your-cert.pem
+```
+
+The certificate is printed to the controller log at startup and is also available via an HTTP GET request to `/v1/cert.pem` on the controller.
+
+### **Secret encryption**
+
+For the encryption, AES-256-GCM is used for the secret’s encryption with a randomly-generated single-use 32 byte session key.
```

Comment:
1. This could probably use some rephrasing (for example starting with "The secret is encrypted..."
2. Same comment regarding the asymmetric key generation, what's the entropy source?